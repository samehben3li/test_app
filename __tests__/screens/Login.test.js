import React from "react";
import renderer from "react-test-renderer";
import Login from "../../screens/Login";
import { MockedProvider } from "@apollo/client/testing";
import AuthProvider from "../../context/authContext";

describe("Login Screen", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <MockedProvider>
          <AuthProvider value={{}}>
            <Login />
          </AuthProvider>
        </MockedProvider>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
