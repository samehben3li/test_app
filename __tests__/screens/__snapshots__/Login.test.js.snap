// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ebecef",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "black",
        "fontSize": 40,
        "marginBottom": 10,
      }
    }
  >
    Welcome
  </Text>
  <View
    style={
      Object {
        "height": 40,
        "width": "80%",
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Email Address"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 7,
        "height": 40,
        "marginTop": 15,
        "paddingHorizontal": 20,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "height": 15,
        "marginVertical": 5,
        "width": "75%",
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 7,
        "height": 40,
        "marginTop": 15,
        "paddingHorizontal": 20,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "height": 15,
        "marginVertical": 5,
        "width": "75%",
      }
    }
  />
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFDE6A",
          "opacity": 0.7,
        },
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 20,
          "width": "80%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "bold",
          "fontSize": 20,
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
